iotdb-client/client-py/iotdb/SessionPool.py

```diff
@@ -0,0 +1,122 @@
+        self.ip = ip
+        if node_urls is None:
+            node_urls = []
+        self.node_urls = node_urls
+        self.user_name = user_name
+        self.password = password
+        self.fetch_size = fetch_size
+        self.time_zone = time_zone
+        self.max_retry = max_retry
+
+
+class SessionPool(object):
+
+    def __init__(self, pool_config: PoolConfig, max_pool_size: int, wait_timeout_in_ms: int):
+        self.__pool_config = pool_config
+        self.__max_pool_size = max_pool_size
+        self.__wait_timeout_in_ms = wait_timeout_in_ms / 1000
+        self.__pool_size = 0
+        self.__queue = Queue(max_pool_size)
+        self.__lock = Lock()
+
+    def __construct_session(self) -> Session:
+        if len(self.__pool_config.node_urls) > 0:
+            return Session.init_from_node_urls(self.__pool_config.node_urls, self.__pool_config.user_name,
+                                               self.__pool_config.password, self.__pool_config.fetch_size,
+                                               self.__pool_config.time_zone)
+
+        else:
+            return Session(self.__pool_config.host, self.__pool_config.ip, self.__pool_config.user_name,
+                           self.__pool_config.password, self.__pool_config.fetch_size, self.__pool_config.time_zone)
+
+    def __poll_session(self) -> Session | None:
+        if self.__queue.empty():
+            return None
+        return self.__queue.get(block=False)
+
+    def get_session(self) -> Session:
+        should_create = False
+        start = time.time()
+
+        session = self.__poll_session()
+        while session is None:
+            self.__lock.acquire()
+            if self.__pool_size < self.__max_pool_size:
+                self.__pool_size += 1
+                should_create = True
+                self.__lock.release()
+                break
+            else:
+                if time.time() - start > self.__wait_timeout_in_ms:
+                    self.__lock.release()
+                    raise TimeoutError("Wait to get session timeout in SessionPool, current pool size: {0}"
+                                       .format(self.__max_pool_size))
+                time.sleep(1)
+            session = self.__poll_session()
+            self.__lock.release()
+
+        if should_create:
+            try:
+                session = self.__construct_session()
+            except Exception as e:
+                self.__lock.acquire()
+                self.__pool_size -= 1
+                self.__lock.release()
+                raise e
+
+        return session
+
+    def put_back(self, session: Session):
+        if session.is_open():
+            self.__queue.put(session)
+        else:
+            self.__lock.acquire()
+            self.__pool_size -= 1
+            self.__lock.release()
+
+    def close(self):
+        while not self.__queue.empty():
+            session = self.__queue.get(block=False)
+            session.close()
```

Comment:
don't need to update `__pool_size`?

Reply:
This closes the sessionPool, so I think it's not necessary to update pool_size.

Reply:
But there is no guarantee that users won't use the closed SessionPool. What if they use it after closing it?

Reply:
I see. I have added a `closed` status for SessionPool